src/urllib3/util/retry.py

```diff
@@ -235,7 +235,7 @@ def __init__(
         self.history = history or ()
         self.respect_retry_after_header = respect_retry_after_header
         self.remove_headers_on_redirect = frozenset(
-            [h.lower() for h in remove_headers_on_redirect]
+            h.lower() for h in remove_headers_on_redirect
```

Comment:
This is technically slower but unlikely to matter in practice

Reply:
Which tool does this change come from? We should file an issue that this construction shouldn't be modified.

Reply:
My eyes failed me, don't know how I read this as a `str.join()` case. This is likely faster.